ada/tests_views.py

```diff
@@ -0,0 +1,95 @@
+from urllib.parse import urljoin
+
+from django.test import TestCase
+from django.urls import reverse
+
+from .models import RedirectUrl
+
+
+class TestViews(TestCase):
+    def test_whatsapp_url_error(self):
+        """
+        Should check that the right template is used for no whatsapp string in url
+        """
+        response = self.client.get(reverse("ada_hook", args=["1"]))
+        self.assertTemplateUsed(response, "index.html")
+
+    def test_name_error(self):
+        """
+        Should check that the right template is used for mis-spelt whatsappid
+        """
+        qs = "?whatsappi=12345"
+        url = urljoin(reverse("ada_hook", args=["1"]), qs)
+        response = self.client.get(url)
+        self.assertTemplateUsed(response, "index.html")
+
+    def test_no_whatsapp_value(self):
+        """
+        Should check that the right template is used if there's no whatsapp value
+        """
+        qs = "?whatsapp="
+        url = urljoin(reverse("ada_hook", args=["1"]), qs)
+        response = self.client.get(url)
+        self.assertTemplateUsed(response, "index.html")
+
+    def test_no_query_string(self):
+        """
+        Should check that the right template is used if there's no whatsapp value
+        """
+        qs = "?"
+        url = urljoin(reverse("ada_hook", args=["1"]), qs)
+        response = self.client.get(url)
+        self.assertTemplateUsed(response, "index.html")
+
+    def test_name_success(self):
+        """
+        Should use the meta refresh template if url is correct
+        """
+        qs = "?whatsappid=12345"
+        url = urljoin(reverse("ada_hook", args=["1"]), qs)
+        response = self.client.get(url)
+        self.assertTemplateUsed(response, "meta_refresh.html")
+
+    def test_special_character(self):
+        """
+        Should use the meta refresh template if url is correct
```

Comment:
This description doesn't match up with what the test is asserting. Special characters in query strings need to be encoded with https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlencode, and if they are encoded properly, then it should just work without error